src/gl/special_case.cpp

```diff
@@ -55,7 +55,18 @@ uint32_t gl_special_case(uint32_t primitivetype, uint32_t vertextype, struct nve
 	if(ff8) current_state.texture_filter = enable_bilinear && vertextype != TLVERTEX && current_state.texture_set;
 	else if (enable_bilinear && (vertextype != TLVERTEX || mode == MODE_MENU || (current_state.texture_set && VREF(texture_set, ogl.gl_set->force_filter)))) current_state.texture_filter = true;
 
-	// some modpath textures have z-sort forced on
+  // since we are no longer disabling all filtering on interal textures, we need to shut it off for the ones that filtering breaks.
+  // don't filter original textures in submarine minigame
+  if (current_state.texture_set && (mode == MODE_SUBMARINE) && !(VREF(texture_set, ogl.external))) // only ff7 has a submarine mode.
+  {
+    current_state.texture_filter = false; // don't filter original submarine textures, this causes thin lines, and original did not. 
+  }
+  // or coaster minigame
+  if (current_state.texture_set && (mode == MODE_COASTER) && !(VREF(texture_set, ogl.external))) // only ff7 has coaster minigame.
+  {
+    current_state.texture_filter = false; // don't filter original coaster textures
+  }
```

Comment:
For both conditions even though we know it's only for ff7, I'd appreciate if we could add `!ff8` nevertheless in the condition check. Additionally, please avoid over commenting ( stuff like `// only ff7 has a submarine mode.` is literally noise while looking at the code ), as well as `// don't filter original ... textures,`, just add notes of what it breaks in case it is set to true. No need to state the obvious :)

Reply:
It seems we have different code documentation views.  Not saying you are wrong. as for leaving out the !ff8 check that's a matter of optimization, with an explanation of why i didn't bother to check. But i see the point. I should be able to refactor that code easily enough.